internal/transport/http2_client.go

```diff
@@ -681,6 +681,8 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (_ *Strea
 	}
 	if t.statsHandler != nil {
 		header, _, _ := metadata.FromOutgoingContextRaw(ctx)
+		header = header.Copy()
```

Comment:
And since this is making a copy, the `header.Copy()` below (when assigning to OutHeader) shouldn't be necessary anymore.

Reply:
Good catch, done.